### PR TITLE
Add details for storage and knative services

### PR DESCRIPTION
**Pull request by skonto**
## Proposed Changes 

- Adds a section for storage integration with Knative Services. Could be expanded with more examples in the feature.

### CODE REVIEW

File: docs/serving/services/storage.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
+# Configuring storage for Knative services
+
+Knative Serving integrates with K8s storage capabilities via supporting a subset of ephemeral storage volumes and via PersistentVolumeClaims(PVCs) volume types.
+In detail Knative Serving supports emptyDir by default, secret, configmap, projection and PVCs volume types.
+For more details on volume configuration check the related [feature flags](../configuration/feature-flags.md) when applicable.
```
Comment (dprotaso) [inline docs/serving/services/storage.md:5]:
```suggestion
By default Serving supports the mounting the [volume types](https://kubernetes.io/docs/concepts/storage/volumes):`emptyDir`, `secret`, `configMap` and `projected`. [PersistentVolumes](https://kubernetes.io/docs/concepts/storage/persistent-volumes/) are supported but require a [feature flag](../configuration/feature-flags.md) to be enabled.
```

File: docs/serving/services/storage.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+Here is an example of using PVCs:
```
Comment (dprotaso) [inline docs/serving/services/storage.md:7]:
```suggestion
Here is an example of using a persistent volume claim with a Knative Service:
```

File: docs/serving/services/storage.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+spec:
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 1Gi
+```
+
+The example assumes that the user has enabled PVC support via the corresponding feature flag.
+
```
Comment (dprotaso) [inline docs/serving/services/storage.md:41]: Can drop this since we mention it above

```suggestion
```

File: docs/serving/services/storage.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+!!! warning
+    Mounting large volumes may add considerable overhead to the application's start up time.
```
Comment (dprotaso) [inline docs/serving/services/storage.md:43]: I'd consider moving this before the example

File: config/nav.yml [hunk 5]
```diff
@@ -131,6 +131,7 @@ nav:
           # TODO: Add security section to docs?
           - Configure resource requests and limits: serving/services/configure-requests-limits-services.md
           - HTTPS redirection: serving/services/http-protocol.md
+          - Configuring storage: serving/services/storage.md
```
Comment (dprotaso) [inline config/nav.yml:134]:
```suggestion
          - Volume Support: serving/services/storage.md
```

**Review by ReToCode — APPROVED** [review]
/lgtm
/approve

**Pull request merged**